### ISSUE TITLE
**BuildPropertiesTest is Failing:**

### ISSUE THREAD

**Issue by jfrairigh** [TEAMMATES/teammates#10982]
**Problem Introduction:**
The build properties component test is failing in both my local master and working branch. My master branch is clean and up to date with Teammates' master. Since the test is failing in both branches, I think it is a configuration issue.

**What I have tried to fix the failing test:**
1.) I have followed Step 3 of https://github.com/TEAMMATES/teammates/blob/master/docs/setting-up.md
2.) I noticed some of the "build.template.properties" files had additional content that the "build.properties" files didn't have so I copied the template files and named the copy "build.properties". 

Does anyone have any insight that will help me with this? 

Here is a screenshot of the test failing: 
![image](https://user-images.githubusercontent.com/67874354/108921954-39641200-75fc-11eb-9f8b-9aef75d46852.png)

_Originally posted by @jfrairigh in https://github.com/TEAMMATES/teammates/issues/10569#issuecomment-784609146_

**Comment by moziliar** [comment]
May I check if you are able start up the backend using `./gradlew appengineRun`? It might be that the thread failed to read in the runtime environment and failed to generate the AppURL, hence failing the test.

**Comment by jfrairigh** [comment]
I started the backend using ./gradlew appengineRun. It looks normal to me. Here is a screenshot of it:
![image](https://user-images.githubusercontent.com/67874354/109265432-29029180-77cc-11eb-96e9-6cdca07d534b.png)

**Comment by moziliar** [comment]
The failing test asks for the server url to be not null; this is generated by the appengine api sdk. Could you check your sdk version? Also, could you try to put some debugging lines in `src/main/java/teammates/common/util/Config.java` around line 129?

**Comment by jfrairigh** [comment]
I have google cloud sdk version 311.0.0. I will look at the Config.java file. Thank you for your help so far.

**Comment by Derek-Hardy** [comment]
@jfrairigh Could you try to delete the existing `build.properties` file and re-run `./gradlew createConfigs`? This will generate the latest config consistent with `build.template.properties`.

Then you can test again whether the same error persists.

**Comment by moziliar** [comment]
Hi @jfrairigh , any update on the problem? Does it still exist?
Consider upgrading the cloud sdk if it is not resolved after the above step (I am on 326.0.0).

**Comment by jfrairigh** [comment]
I tried @Derek-Hardy 's suggestion, but it didn't fix the problem. I did notice only four out of the five build.properties files were regenerated by ./gradlew createConfigs, which I thought was interesting. I ran the test and it failed. Then, I manually made a copy of the 5th build.properties file, which is the one in teammates/build/exploded-repository/WEB-INF/classes. Ran the test again and it failed again. I will try upgrading the sdk. Hopefully, that will work.
